modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,220 @@
+
+    const option = await this.kvs.get(key, `[${index}]`)
+    if (!option) {
+      return
+    }
+
+    await this.kvs.delete(key)
+
+    const { type, label, value } = option
+    return {
+      type,
+      text: type === 'say_something' ? value : label,
+      payload: value
+    }
+  }
+
+  async handleOutgoingEvent(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    const { type, payload } = event
+    const { __buttons, __dropdown } = payload.metadata as sdk.Content.Metadata
+
+    if (__buttons) {
+      await this.sendChoices(event)
+    } else if (__dropdown) {
+      await this.sendDropdown(event)
+    } else if (type === 'text') {
+      await this.sendText(event)
+    } else if (type === 'image') {
+      await this.sendImage(event)
+    } else if (type === 'carousel') {
+      await this.sendCarousel(event)
+    }
+    next(undefined, false)
+  }
+
+  async sendText(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.text
+    })
+  }
+
+  async sendImage(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.title,
+      mediaUrl: [`${event.payload.BOT_URL}${event.payload.image}`]
+    })
+  }
+
+  async sendCarousel(event: sdk.IO.Event) {
+    for (const { subtitle, title, image, actions } of event.payload.items) {
+      const body = `${title}\n\n${subtitle ? subtitle : ''}`
+
+      const options: MessageOption[] = []
+      for (const { action, title, payload, url, text } of actions) {
+        if (action === 'Open URL') {
+          options.push({ label: `${title} : ${url}`, value: undefined, type: 'url' })
+        } else if (action === 'Postback') {
+          options.push({ label: title, value: payload, type: 'postback' })
+        } else if (action === 'Say something') {
+          options.push({ label: title, value: text, type: 'say_something' })
+        }
+      }
+
+      const args = { mediaUrl: image ? [`${event.payload.BOT_URL}${image}`] : undefined }
+      await this.sendOptions(event, body, args, options)
+    }
+  }
+
+  async sendChoices(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.metadata.__buttons.map(x => ({
+      label: x.title,
+      value: x.payload,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.text, {}, options)
+  }
+
+  async sendDropdown(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.metadata.__dropdown.map(x => ({
+      label: x.label,
+      value: x.value,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.message, {}, options)
+  }
+
+  async sendOptions(event: sdk.IO.Event, text: string, args: any, options: MessageOption[]) {
+    const body = `${text}\n\n${options.map(({ label }, idx) => `${idx + 1}. ${label}`).join('\n')}`
+
+    await this.kvs.set(this.getKvsKey(event.target, event.threadId), options, undefined, '10m')
+
+    await this.sendMessage(event, { ...args, body })
```

Comment:
Can be merged with the sendChoices method (sendDropdown is not necessary)

Reply:
sendCarousel also uses this method